src/Workspaces/Core/MSBuild/MSBuild/CSharp/CSharpProjectFile.cs

```diff
@@ -0,0 +1,440 @@
+                this.Log);
+        }
+
+        private ImmutableArray<string> FixPlatform(ImmutableArray<string> commandLineArgs)
+        {
+            string platform = null, target = null;
+            var platformIndex = -1;
+
+            for (int i = 0; i < commandLineArgs.Length; i++)
+            {
+                var arg = commandLineArgs[i];
+
+                if (platform != null && arg.StartsWith("/platform:", StringComparison.OrdinalIgnoreCase))
+                {
+                    platform = arg.Substring("/platform:".Length);
+                    platformIndex = i;
+                }
+                else if (target != null && arg.StartsWith("/target:", StringComparison.OrdinalIgnoreCase))
+                {
+                    target = arg.Substring("/target:".Length);
+                }
+
+                if (platform != null && target != null)
+                {
+                    break;
+                }
+            }
+
+            if (string.Equals("anycpu32bitpreferred", platform, StringComparison.OrdinalIgnoreCase)
+                && (string.Equals("library", target, StringComparison.OrdinalIgnoreCase)
+                    || string.Equals("module", target, StringComparison.OrdinalIgnoreCase)
+                    || string.Equals("winmdobj", target, StringComparison.OrdinalIgnoreCase)))
+            {
+                return commandLineArgs.SetItem(platformIndex, "/platform:anycpu");
+            }
+
+            return commandLineArgs;
+        }
+
+        private ImmutableArray<string> GetAliases(MSB.Framework.ITaskItem item)
+        {
+            var aliasesText = item.GetMetadata("Aliases");
+
+            if (string.IsNullOrEmpty(aliasesText))
+            {
+                return ImmutableArray<string>.Empty;
+            }
+
+            return ImmutableArray.CreateRange(aliasesText.Split(new char[] { ' ', ',' }, StringSplitOptions.RemoveEmptyEntries));
+        }
+
+        private ImmutableArray<string> ReadCommandLineArguments(MSB.Execution.ProjectInstance project)
+        {
+            var builder = ImmutableArray.CreateBuilder<string>();
+
+            var additionalFiles = this.GetAdditionalFilesFromModel(project);
+            if (additionalFiles != null)
+            {
+                foreach (var additionalFile in additionalFiles)
+                {
+                    builder.Add("/additionalfile:\"" + this.GetDocumentFilePath(additionalFile) + "\"");
+                }
+            }
+
+            var allowUnsafeBlocks = project.ReadPropertyBool("AllowUnsafeBlocks");
+            if (allowUnsafeBlocks)
+            {
+                builder.Add("/unsafe");
+            }
+
+            var analyzers = this.GetAnalyzerReferencesFromModel(project);
+            if (analyzers != null)
+            {
+                foreach (var analyzer in analyzers)
+                {
+                    builder.Add("/analyzer:\"" + this.GetDocumentFilePath(analyzer) + "\"");
```

Comment:
lots of string seems can be converted to new string interpolation

Reply:
good point. Will do.